File: server/src/entity/atm.ts
import { BaseEntity, Column, Entity, Index, PrimaryColumn } from "typeorm";

export const enum Bank {
  piraeus = "piraeus",
  national = "national",
  alpha = "alpha",
}

@Entity()
@Index(["id", "bank"], { unique: true })
@Index(["bank"])
export class ATM extends BaseEntity {
  @PrimaryColumn({
    nullable: false,
  })
  id!: string;

  @Column()
  bank!: string;

  @Column()
  name!: string;

  @Column({
    type: "float",
  })
  latitude!: number;

  @Column()
  fullAddress!: string;

  @Column({
    type: "float",
  })
  longitude!: number;

  @Column()
  address!: string;

  @Column()
  postalCode!: string;

  @Column()
  municipality!: string;

  @Column()
  prefecture!: string;

  @Column()
  isOffsite!: boolean;

  @Column()
  hasWifi!: boolean;

  @Column()
  workingHours?: string;

  getFullId() {
    return `${this.bank}_${this.id}`;
  }
}

File: app/lib/models/atm.dart
import 'package:json_annotation/json_annotation.dart';

part 'atm.g.dart';

@JsonSerializable()
class ATM {
  final String bank;
  final String name;
  final String fullAddress;

  final double latitude;
  final double longitude;

  final bool isOffsite;
  final bool hasWifi;

  final String workingHours;

  ATM(
    this.bank,
    this.name,
    this.fullAddress,
    this.latitude,
    this.longitude,
    this.isOffsite,
    this.hasWifi,
    this.workingHours,
  );

  factory ATM.fromJson(Map<String, dynamic> json) => _$ATMFromJson(json);

  Map<String, dynamic> toJson() => _$ATMToJson(this);
}
